File: 16-spring-boot-flux/src/main/java/com/gzz/user/UserDao.java
package com.gzz.user;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.BeanPropertyRowMapper;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;
/**
 * @author https://www.jianshu.com/u/3bd57d5f1074
 * @date 2019-12-24 10:50:00
 */
@Repository
public class UserDao {
	@Autowired
	protected JdbcTemplate jdbcTemplate;// jdbc模版类

	public List<User> queryList() {
		String sql = "select * from sys_person";
		return jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(User.class));
	}

	public User findById(String id) {
		String sql = "select * from sys_person where id=?";
		String param[] = { id };
		return jdbcTemplate.queryForObject(sql, param, new BeanPropertyRowMapper<>(User.class));
	}

}

File: 99-spring-boot-kotlin/src/main/java/com/gzz/mvc/UserDao.kt
package com.gzz.mvc


import org.springframework.stereotype.Repository

import org.springframework.jdbc.core.BeanPropertyRowMapper
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import com.gzz.base.BaseDao
import com.gzz.base.Page
import java.lang.StringBuffer

@Component
class UserDao : BaseDao() {
	fun queryList(cond: UserCond): List<User> {
		var sb: StringBuffer = StringBuffer();
		sb.append("select * from sys_user where 1=1 ")
		sb.append(cond.getCondition())
		return jdbcTemplate.query(sb.toString(), cond.getArray(), BeanPropertyRowMapper<User>(User::class.java))
	}

	fun queryPage(cond: UserCond): Page<User> {
		var sb: StringBuffer = StringBuffer();
		sb.append("select * from sys_user where 1=1 ")
		sb.append(cond.getCondition())
		return queryPage(sb.toString(), cond, User::class.java)
	}

	fun save(user: User): Int {
		var sql: String = "insert into sys_user (name,password,login_id) values (?,?,?)"
		var params = arrayOf<Any>(user.name, user.password, user.login_id);
		return jdbcTemplate.update(sql, params)
	}

	fun delete(id: Long): Int {
		var sql: String = "delete from sys_user where id=?"
		return jdbcTemplate.update(sql, arrayOf<Any>(id))
	}

	fun findById(id: Long): User {
		var sql: String = "select *  from sys_user where id=?"
		return jdbcTemplate.queryForObject(sql, arrayOf<Any>(id), BeanPropertyRowMapper<User>(User::class.java))
	}
}